sound/soc/sof/intel/hda-pcm.c

```diff
@@ -89,13 +89,19 @@ int hda_dsp_pcm_hw_params(struct snd_sof_dev *sdev,
 	struct hdac_ext_stream *stream = stream_to_hdac_ext_stream(hstream);
 	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
 	struct snd_dma_buffer *dmab;
+	struct snd_sof_pcm *spcm;
 	int ret;
 	u32 size, rate, bits;
 
 	size = params_buffer_bytes(params);
 	rate = get_mult_div(sdev, params_rate(params));
 	bits = get_bits(sdev, params_width(params));
 
+	spcm = snd_sof_find_spcm_dai(sdev, substream->private_data);
+	if (spcm)
+		INIT_WORK(&spcm->stream[substream->stream].period_elapsed_work,
+			  sof_pcm_period_elapsed_work);
+
```

Comment:
ok, I stop here.
These changes should either be dropped or submitted to the team.
This development flow is not ok in my book.